src/catalog/function_catalog.cpp

```diff
@@ -0,0 +1,355 @@
+    }
+  }
+
+  else {
+    LOG_DEBUG("No function by that name!");
+  }
+  
+  return function_info;
+}
+
+ ResultType  FunctionCatalog::InsertFunction(const std::string &proname,
+                          UNUSED_ATTRIBUTE oid_t pronamespace, UNUSED_ATTRIBUTE oid_t proowner, 
+                          oid_t prolang, UNUSED_ATTRIBUTE float procost,
+                         UNUSED_ATTRIBUTE float prorows,UNUSED_ATTRIBUTE oid_t provariadic,UNUSED_ATTRIBUTE bool proisagg,
+                         UNUSED_ATTRIBUTE bool proiswindow,UNUSED_ATTRIBUTE bool prosecdef,UNUSED_ATTRIBUTE bool proleakproof,
+                         UNUSED_ATTRIBUTE bool proisstrict,UNUSED_ATTRIBUTE bool proretset,UNUSED_ATTRIBUTE char provolatile,
+                          int pronargs,UNUSED_ATTRIBUTE int pronargdefaults, oid_t prorettype,
+                          std::vector<type::Type::TypeId> proargtypes, 
+                         UNUSED_ATTRIBUTE std::vector<type::Type::TypeId> proallargtypes,
+                         UNUSED_ATTRIBUTE std::vector<int> proargmodes, 
+                          std::vector<std::string> proargnames, 
+                         UNUSED_ATTRIBUTE std::vector<type::Type::TypeId> proargdefaults, 
+                          std::vector<std::string> prosrc_bin, 
+                         UNUSED_ATTRIBUTE std::vector<std::string> proconfig, UNUSED_ATTRIBUTE std::vector<int> aclitem,
+                          type::AbstractPool *pool,
+                          concurrency::Transaction *txn){
+      
+      std::unique_ptr<storage::Tuple> tuple(new storage::Tuple(catalog_table_->GetSchema(), true));
+
+      //generate a unique oid
+      auto oid = GetNextOid(); 
+      auto val0 = type::ValueFactory::GetIntegerValue(oid);
+      auto val1 = type::ValueFactory::GetVarcharValue(proname, nullptr);
+      auto val2 = type::ValueFactory::GetNullValueByType(type::Type::INTEGER);
+      auto val3 = type::ValueFactory::GetIntegerValue(prolang);
+      auto val4 = type::ValueFactory::GetNullValueByType(type::Type::DECIMAL);
+      auto val5 = type::ValueFactory::GetNullValueByType(type::Type::DECIMAL);
+      auto val6 = type::ValueFactory::GetNullValueByType(type::Type::INTEGER);
+      auto val7 = type::ValueFactory::GetNullValueByType(type::Type::BOOLEAN);
+      auto val8 = type::ValueFactory::GetNullValueByType(type::Type::BOOLEAN);
+      auto val9 = type::ValueFactory::GetNullValueByType(type::Type::BOOLEAN);
+      auto val10 = type::ValueFactory::GetNullValueByType(type::Type::BOOLEAN);
+      auto val11 = type::ValueFactory::GetNullValueByType(type::Type::BOOLEAN);
+      auto val12 = type::ValueFactory::GetNullValueByType(type::Type::BOOLEAN);
+      auto val13 = type::ValueFactory::GetNullValueByType(type::Type::VARCHAR);
+      auto val14 = type::ValueFactory::GetIntegerValue(pronargs);
+      auto val15 = type::ValueFactory::GetNullValueByType(type::Type::INTEGER);
+      auto val16 = type::ValueFactory::GetIntegerValue(prorettype);
```

Comment:
Currently we don't really use all the columns of the pg_proc catalog, but in the future it might be useful. This code can be easily adapted to have this addition.